File: sales.py
import numpy as np
import pandas as pd

# 批次讀取 Q1~Q4，自動加入 quarter 欄位並合併成一張資料表
dfs = [
    pd.read_csv(f"Q{i}.csv").assign(quarter=f"Q{i}")
    for i in range(1, 5)
]
sales_all = pd.concat(dfs, ignore_index=True)

# 定義高價值用戶
sales_all['value'] = np.where(sales_all['sale_amount'] >= 100, "高价值", "普通")

File: sales.R
library(tidyverse)

# 批次讀取 Q1~Q4，自動加入 quarter 欄位並合併成一張資料表
sales_all <- map_dfr(1:4, \(i) {
  q_name <- paste0("Q", i)
  read.csv(paste0(q_name, ".csv")) %>%
    mutate(quarter = q_name)
})
#定义高价值用户
sales_all$value <- ifelse(sales_all$sale_amount >= 100, "高价值", "普通")
